### ISSUE TITLE
Packages that needs fix without deprecated APIs for OpenSSL

### ISSUE THREAD

**Issue by frymezim** [openwrt/packages#7367]
@neheb 
Their are 3 packages that I know causing breakage on compilation without deprecated APIs. Can you provide a patch for this?
lighttpd
wget
libssh2

Also is there a easier way to apply the pull request on my own buildroot because I was able to do this on openwrt git, editing the .git/config and use git merge origin/pr/####.

**Comment by neheb** [comment]
Is this on OpenSSL 1.1.x? I've only been testing 1.0.2. Wget patch to fix that has been submitted upstream.

As far as pull requests go, I used to use git-pr for that. Just git-pr ### and git checkout original followed by git cherry-pick.

Nowadays I add the branch directly using git remote and cherry pick that.

**Comment by frymezim** [comment]
Yes it's OpenSSL 1.1.x been trying that patch and those three have problems with compilation.

**Comment by neheb** [comment]
I tried fixing wget. Didn't work. Some autoconf problem.

**Comment by frymezim** [comment]
Ok cool, so I want to test those patches can you give me an example how to merge your pull request on my  build environment from the /home/username/openwrt/ folder thanks!

**Comment by frymezim** [comment]
Another issue I found that needed OpenSSL deprecated API.

```
make[3]: Entering directory '/home/dkmori/openwrt/feeds/packages/lang/python/python-pip-conf'
make[3]: Nothing to be done for 'compile'.
make[3]: Leaving directory '/home/dkmori/openwrt/feeds/packages/lang/python/python-pip-conf'
time: package/feeds/packages/python-pip-conf/compile#0.07#0.02#0.09
make[3]: Entering directory '/home/dkmori/openwrt/feeds/packages/lang/python/python'
make[3]: Leaving directory '/home/dkmori/openwrt/feeds/packages/lang/python/python'
time: package/feeds/packages/python/host-compile#0.15#0.15#0.32
make[3]: Entering directory '/home/dkmori/openwrt/feeds/packages/lang/python/python'
mkdir -p /home/dkmori/openwrt/bin/targets/kirkwood/generic/packages /home/dkmori/openwrt/build_dir/target-arm_xscale_musl_eabi/Python-2.7.15/ipkg-arm_xscale/python-openssl/CONTROL /home/dkmori/openwrt/staging_dir/target-arm_xscale_musl_eabi/pkginfo
find /home/dkmori/openwrt/build_dir/target-arm_xscale_musl_eabi/Python-2.7.15/ipkg-install -name "*\.exe" | xargs rm -f
/usr/bin/env bash ../python-package-install.sh "2" "/home/dkmori/openwrt/build_dir/target-arm_xscale_musl_eabi/Python-2.7.15/ipkg-install" "/home/dkmori/openwrt/build_dir/target-arm_xscale_musl_eabi/Python-2.7.15/ipkg-arm_xscale/python-openssl" "/home/dkmori/openwrt/staging_dir/hostpkg/bin/python2.7" "" "$V_PyPackage_python_openssl_filespec"
copying: '/usr/lib/python2.7/lib-dynload/_hashlib.so'
File not found '/home/dkmori/openwrt/build_dir/target-arm_xscale_musl_eabi/Python-2.7.15/ipkg-install/usr/lib/python2.7/lib-dynload/_ssl.so'
process filespec error-ed
make[3]: *** [Makefile:304: /home/dkmori/openwrt/bin/packages/arm_xscale/packages/python-openssl_2.7.15-1_arm_xscale.ipk] Error 1
make[3]: Leaving directory '/home/dkmori/openwrt/feeds/packages/lang/python/python'
time: package/feeds/packages/python/compile#0.17#0.10#0.24
make[2]: *** [package/Makefile:109: package/feeds/packages/python/compile] Error 2
make[2]: Leaving directory '/home/dkmori/openwrt'
make[1]: *** [package/Makefile:103: /home/dkmori/openwrt/staging_dir/target-arm_xscale_musl_eabi/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/dkmori/openwrt'
make: *** [/home/dkmori/openwrt/include/toplevel.mk:218: world] Error 2
```

**Comment by neheb** [comment]
Ha. Ha. Ha.

Python is a mess. I have a patch in my local tree that fixes Python itself. 2 and 3 actually. python-cryptography requires such massive patch that it's not even worth it. I don't expect python-openssl to be much better.

It's probably better to add a dependency on OPENSSL_WITH_DEPRECATED.

**Comment by commodo** [comment]
@frymezim 
can you provide a `.config` file that reproduces this ? and some details
i can't seem to reproduce it on my end

**Comment by neheb** [comment]
@commodo compile without deprecated APIs on OpenSSL.

**Comment by neheb** [comment]
@frymezim https://github.com/neheb/packages/commits/pydep

**Comment by commodo** [comment]
Python still seems to build for me.
Tried it on a Marvel Kirkwood, since that looks like it was in the report.

SSL section looks like this:
```
#
# SSL
#
# CONFIG_PACKAGE_libgnutls is not set
# CONFIG_PACKAGE_libmbedtls is not set
CONFIG_PACKAGE_libopenssl=y
CONFIG_OPENSSL_WITH_EC=y
# CONFIG_OPENSSL_WITH_EC2M is not set
# CONFIG_OPENSSL_WITH_SSL3 is not set
# CONFIG_OPENSSL_WITH_DEPRECATED is not set
CONFIG_OPENSSL_NO_DEPRECATED=y
# CONFIG_OPENSSL_WITH_DTLS is not set
# CONFIG_OPENSSL_WITH_COMPRESSION is not set
CONFIG_OPENSSL_WITH_NPN=y
CONFIG_OPENSSL_WITH_PSK=y
CONFIG_OPENSSL_WITH_SRP=y
# CONFIG_OPENSSL_HARDWARE_SUPPORT is not set
# CONFIG_OPENSSL_OPTIMIZE_SPEED is not set
# CONFIG_OPENSSL_ENGINE_CRYPTO is not set
# CONFIG_PACKAGE_libwolfssl is not set
```

**Comment by neheb** [comment]
Oh right. He's talking about the OpenSSL patch for 1.1 on the main repo.

**Comment by neheb** [comment]
ACK. The original code is totally wrong.

**Comment by frymezim** [comment]
Python Patches are working fine.
Compiled & Tested NSA325 / Kirkwood

@neheb 
Another issue I found is for OpenSSH FYI this need I'm using OpenSSL 1.1.1
```
make[5]: Leaving directory '/home/dkmori/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/openssh-without-pam/openssh-7.9p1/openbsd-compat'
arm-openwrt-linux-muslgnueabi-gcc -O3 -pipe -mcpu=cortex-a15 -mfpu=neon-vfpv4 -mfloat-abi=hard -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/home/dkmori/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/openssh-without-pam/openssh-7.9p1=openssh-7.9p1 -Wformat -Werror=format-security -fpic -fstack-protector-strong -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -pipe -Wall -Wpointer-arith -Wuninitialized -Wsign-compare -Wformat-security -Wsizeof-pointer-memaccess -Wno-pointer-sign -Wno-unused-result -fno-strict-aliasing -ftrapv -fno-builtin-memset -fstack-protector-strong -fPIE   -I. -I. -I/home/dkmori/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/usr/include -I/home/dkmori/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/include -I/home/dkmori/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.2.0_musl_eabi/usr/include -I/home/dkmori/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.2.0_musl_eabi/include/fortify -I/home/dkmori/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.2.0_musl_eabi/include  -D_XOPEN_SOURCE=600 -D_BSD_SOURCE -D_DEFAULT_SOURCE -DSSHDIR=\"/etc/ssh\" -D_PATH_SSH_PROGRAM=\"/usr/bin/ssh\" -D_PATH_SSH_ASKPASS_DEFAULT=\"/usr/lib/ssh-askpass\" -D_PATH_SFTP_SERVER=\"/usr/lib/sftp-server\" -D_PATH_SSH_KEY_SIGN=\"/usr/lib/ssh-keysign\" -D_PATH_SSH_PKCS11_HELPER=\"/usr/lib/ssh-pkcs11-helper\" -D_PATH_SSH_PIDDIR=\"/var/run\" -D_PATH_PRIVSEP_CHROOT_DIR=\"/var/empty\" -DHAVE_CONFIG_H -c ssh.c -o ssh.o
ssh.c: In function 'main':
ssh.c:819:8: warning: implicit declaration of function 'SSLeay_version'; did you mean 'RSA_get_version'? [-Wimplicit-function-declaration]
        SSLeay_version(SSLEAY_VERSION)
        ^~~~~~~~~~~~~~
        RSA_get_version
ssh.c:819:23: error: 'SSLEAY_VERSION' undeclared (first use in this function); did you mean 'SSH_VERSION'?
        SSLeay_version(SSLEAY_VERSION)
                       ^~~~~~~~~~~~~~
                       SSH_VERSION
ssh.c:819:23: note: each undeclared identifier is reported only once for each function it appears in
ssh.c:1040:2: warning: implicit declaration of function 'OpenSSL_add_all_algorithms'; did you mean 'ssh_OpenSSL_add_all_algorithms'? [-Wimplicit-function-declaration]
  OpenSSL_add_all_algorithms();
  ^~~~~~~~~~~~~~~~~~~~~~~~~~
  ssh_OpenSSL_add_all_algorithms
ssh.c:1041:2: warning: implicit declaration of function 'ERR_load_crypto_strings'; did you mean 'ERR_load_DH_strings'? [-Wimplicit-function-declaration]
  ERR_load_crypto_strings();
  ^~~~~~~~~~~~~~~~~~~~~~~
  ERR_load_DH_strings
make[4]: *** [Makefile:160: ssh.o] Error 1
make[4]: Leaving directory '/home/dkmori/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/openssh-without-pam/openssh-7.9p1'
make[3]: *** [Makefile:276: /home/dkmori/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/openssh-without-pam/openssh-7.9p1/.built] Error 2
make[3]: Leaving directory '/home/dkmori/openwrt/feeds/packages/net/openssh'
time: package/feeds/packages/openssh/without-pam/compile#37.22#12.42#45.56
make[2]: *** [package/Makefile:109: package/feeds/packages/openssh/compile] Error 2
make[2]: Leaving directory '/home/dkmori/openwrt'
make[1]: *** [package/Makefile:103: /home/dkmori/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/dkmori/openwrt'
make: *** [/home/dkmori/openwrt/include/toplevel.mk:218: world] Error 2
```

**Comment by neheb** [comment]
Most parts of the OpenSSH stuff have been fixed upstream. Just needs to be backported. Or wait for a new version to be released.

**Comment by tripolar** [comment]
@frymezim this code is part of 7.9_p1

**Comment by frymezim** [comment]
@neheb 
I got an error in wget

```
make[3]: Entering directory '/home/test/openwrt/feeds/packages/net/wget'
rm -f /home/test/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/wget-ssl/wget-1.19.5/.configured_*
rm -f /home/test/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/stamp/.wget.ssl_installed
(cd /home/test/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/wget-ssl/wget-1.19.5/./; if [ -x ./configure ]; then find /home/test/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/wget-ssl/wget-1.19.5/ -name config.guess | xargs -r chmod u+w; find /home/test/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/wget-ssl/wget-1.19.5/ -name config.guess | xargs -r -n1 cp --remove-destination /home/test/openwrt/scripts/config.guess; find /home/test/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/wget-ssl/wget-1.19.5/ -name config.sub | xargs -r chmod u+w; find /home/test/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/wget-ssl/wget-1.19.5/ -name config.sub | xargs -r -n1 cp --remove-destination /home/test/openwrt/scripts/config.sub; AR="arm-openwrt-linux-muslgnueabi-gcc-ar" AS="arm-openwrt-linux-muslgnueabi-gcc -c -O3 -pipe -mcpu=cortex-a15 -mfpu=neon-vfpv4 -mfloat-abi=hard -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/home/test/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/wget-ssl/wget-1.19.5=wget-1.19.5 -Wformat -Werror=format-security -fpic -fstack-protector-strong -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro" LD=arm-openwrt-linux-muslgnueabi-ld NM="arm-openwrt-linux-muslgnueabi-gcc-nm" CC="arm-openwrt-linux-muslgnueabi-gcc" GCC="arm-openwrt-linux-muslgnueabi-gcc" CXX="arm-openwrt-linux-muslgnueabi-g++" RANLIB="arm-openwrt-linux-muslgnueabi-gcc-ranlib" STRIP=arm-openwrt-linux-muslgnueabi-strip OBJCOPY=arm-openwrt-linux-muslgnueabi-objcopy OBJDUMP=arm-openwrt-linux-muslgnueabi-objdump SIZE=arm-openwrt-linux-muslgnueabi-size CFLAGS="-O3 -pipe -mcpu=cortex-a15 -mfpu=neon-vfpv4 -mfloat-abi=hard -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/home/test/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/wget-ssl/wget-1.19.5=wget-1.19.5 -Wformat -Werror=format-security -fpic -fstack-protector-strong -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro " CXXFLAGS="-O3 -pipe -mcpu=cortex-a15 -mfpu=neon-vfpv4 -mfloat-abi=hard -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -fmacro-prefix-map=/home/test/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/wget-ssl/wget-1.19.5=wget-1.19.5 -Wformat -Werror=format-security -fpic -fstack-protector-strong -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro " CPPFLAGS="-I/home/test/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/usr/include -I/home/test/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/include -I/home/test/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.2.0_musl_eabi/usr/include -I/home/test/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.2.0_musl_eabi/include/fortify -I/home/test/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.2.0_musl_eabi/include " LDFLAGS="-L/home/test/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/usr/lib -L/home/test/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/lib -L/home/test/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.2.0_musl_eabi/usr/lib -L/home/test/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.2.0_musl_eabi/lib -fpic -specs=/home/test/openwrt/include/hardened-ld-pie.specs -znow -zrelro "  ac_cv_header_uuid_uuid_h=no  ./configure --target=arm-openwrt-linux --host=arm-openwrt-linux --build=x86_64-pc-linux-gnu --program-prefix="" --program-suffix="" --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --libexecdir=/usr/lib --sysconfdir=/etc --datadir=/usr/share --localstatedir=/var --mandir=/usr/man --infodir=/usr/info --disable-nls  --disable-rpath --disable-iri --with-included-libunistring --without-libuuid --without-libpsl --with-ssl=openssl --with-libssl-prefix="/home/test/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/usr" ; fi; )
configure: loading site script /home/test/openwrt/include/site/arm
configure: configuring for GNU Wget 1.19.5
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for arm-openwrt-linux-strip... arm-openwrt-linux-muslgnueabi-strip
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... arm-openwrt-linux-gnu
checking whether make supports nested variables... (cached) yes
checking for arm-openwrt-linux-gcc... arm-openwrt-linux-muslgnueabi-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether arm-openwrt-linux-muslgnueabi-gcc accepts -g... yes
checking for arm-openwrt-linux-muslgnueabi-gcc option to accept ISO C89... none needed
checking whether arm-openwrt-linux-muslgnueabi-gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of arm-openwrt-linux-muslgnueabi-gcc... gcc3
checking how to run the C preprocessor... arm-openwrt-linux-muslgnueabi-gcc -E
checking for grep that handles long lines and -e... /home/test/openwrt/staging_dir/host/bin/grep
checking for egrep... /home/test/openwrt/staging_dir/host/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking for Minix Amsterdam compiler... no
checking for arm-openwrt-linux-ar... arm-openwrt-linux-muslgnueabi-gcc-ar
checking for _LARGEFILE_SOURCE value needed for large files... no
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for a Python interpreter with version >= 3.0... python3
checking for python3... /usr/bin/python3
checking for python3 version... 3.6
checking for python3 platform... linux
checking for python3 script directory... ${prefix}/lib/python3.6/site-packages
checking for python3 extension module directory... ${exec_prefix}/lib/python3.6/site-packages
checking whether NLS is requested... no
checking for msgfmt... /home/test/openwrt/staging_dir/hostpkg/bin/msgfmt
checking for gmsgfmt... /home/test/openwrt/staging_dir/hostpkg/bin/msgfmt
checking for xgettext... /home/test/openwrt/staging_dir/hostpkg/bin/xgettext
checking for msgmerge... /home/test/openwrt/staging_dir/hostpkg/bin/msgmerge
checking for ld... arm-openwrt-linux-muslgnueabi-ld
checking if the linker (arm-openwrt-linux-muslgnueabi-ld) is GNU ld... yes
checking for shared library run path origin... done
checking host CPU and C ABI... armhf
checking for the common suffixes of directories in the library search path... lib,lib
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyCurrent... no
checking whether to use NLS... no
checking for arm-openwrt-linux-ranlib... arm-openwrt-linux-muslgnueabi-gcc-ranlib
checking for flex... flex
checking lex output file root... lex.yy
checking lex library... none needed
checking whether yytext is a pointer... no
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for working volatile... yes
checking for ANSI C header files... (cached) yes
checking for special C compiler options needed for large files... (cached) no
checking for _FILE_OFFSET_BITS value needed for large files... (cached) no
checking size of off_t... (cached) 8
checking for stdbool.h that conforms to C99... yes
checking for _Bool... yes
checking for unistd.h... (cached) yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking termios.h usability... yes
checking termios.h presence... yes
checking for termios.h... yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/select.h usability... yes
checking sys/select.h presence... yes
checking for sys/select.h... yes
checking utime.h usability... yes
checking utime.h presence... yes
checking for utime.h... yes
checking sys/utime.h usability... no
checking sys/utime.h presence... no
checking for sys/utime.h... no
checking for stdint.h... (cached) yes
checking for inttypes.h... (cached) yes
checking pwd.h usability... yes
checking pwd.h presence... yes
checking for pwd.h... yes
checking wchar.h usability... yes
checking wchar.h presence... yes
checking for wchar.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking whether h_errno is declared... yes
checking size of short... (cached) 2
checking size of int... (cached) 4
checking size of long... (cached) 4
checking size of long long... (cached) 8
checking size of void *... (cached) 4
checking for size_t... yes
checking for pid_t... yes
checking for uint32_t... yes
checking for uintptr_t... yes
checking for intptr_t... yes
checking for int64_t... yes
checking for sig_atomic_t... yes
checking for sys/socket.h... yes
checking for arpa/inet.h... yes
checking for features.h... yes
checking for unistd.h... (cached) yes
checking for wctype.h... yes
checking for sys/stat.h... (cached) yes
checking for netdb.h... yes
checking for netinet/in.h... yes
checking for getopt.h... yes
checking for sys/cdefs.h... yes
checking for stdio_ext.h... yes
checking for termios.h... (cached) yes
checking for sys/time.h... (cached) yes
checking for iconv.h... yes
checking for limits.h... yes
checking for wchar.h... (cached) yes
checking for stdint.h... (cached) yes
checking for inttypes.h... (cached) yes
checking for crtdefs.h... no
checking for langinfo.h... yes
checking for xlocale.h... no
checking for sys/mman.h... yes
checking for sys/select.h... (cached) yes
checking for sys/param.h... yes
checking for malloc.h... yes
checking for spawn.h... yes
checking for strings.h... (cached) yes
checking for sys/file.h... yes
checking for sys/ioctl.h... (cached) yes
checking for sys/uio.h... yes
checking for sys/wait.h... yes
checking for utime.h... (cached) yes
checking for stdlib.h... (cached) yes
checking whether the preprocessor supports include_next... yes
checking whether system header files limit the line length... no
checking whether <sys/socket.h> is self-contained... yes
checking for shutdown... yes
checking whether <sys/socket.h> defines the SHUT_* macros... yes
checking for struct sockaddr_storage... yes
checking for sa_family_t... yes
checking for struct sockaddr_storage.ss_family... yes
checking whether socket is declared without a macro... yes
checking whether connect is declared without a macro... yes
checking whether accept is declared without a macro... yes
checking whether bind is declared without a macro... yes
checking whether getpeername is declared without a macro... yes
checking whether getsockname is declared without a macro... yes
checking whether getsockopt is declared without a macro... yes
checking whether listen is declared without a macro... yes
checking whether recv is declared without a macro... yes
checking whether send is declared without a macro... yes
checking whether recvfrom is declared without a macro... yes
checking whether sendto is declared without a macro... yes
checking whether setsockopt is declared without a macro... yes
checking whether shutdown is declared without a macro... yes
checking whether accept4 is declared without a macro... yes
checking for working alloca.h... yes
checking for alloca... yes
checking for C/C++ restrict keyword... __restrict
checking whether <wchar.h> uses 'inline' correctly... yes
checking for btowc... yes
checking for _set_invalid_parameter_handler... no
checking for fcntl... yes
checking for symlink... yes
checking for flock... yes
checking for isblank... yes
checking for iswctype... yes
checking for mbsrtowcs... yes
checking for mempcpy... yes
checking for wmemchr... yes
checking for wmemcpy... yes
checking for wmempcpy... no
checking for futimens... yes
checking for getdelim... yes
checking for getdtablesize... yes
checking for __fsetlocking... yes
checking for tcgetattr... yes
checking for tcsetattr... yes
checking for getprogname... no
checking for getexecname... no
checking for gettimeofday... (cached) yes
checking for nanotime... no
checking for iswcntrl... yes
checking for iswblank... yes
checking for link... yes
checking for lstat... (cached) yes
checking for mbsinit... yes
checking for mbrtowc... yes
checking for mprotect... yes
checking for mkostemp... yes
checking for mkstemp... yes
checking for tzset... yes
checking for nl_langinfo... yes
checking for pipe... yes
checking for pipe2... yes
checking for secure_getenv... no
checking for getuid... yes
checking for geteuid... yes
checking for getgid... yes
checking for getegid... yes
checking for sigaction... yes
checking for sigaltstack... yes
checking for siginterrupt... yes
checking for snprintf... yes
checking for strdup... yes
checking for strerror_r... yes
checking for __xpg_strerror_r... yes
checking for catgets... yes
checking for strndup... yes
checking for strptime... yes
checking for localtime_r... yes
checking for timegm... yes
checking for utime... yes
checking for futimes... yes
checking for futimesat... yes
checking for utimensat... yes
checking for lutimes... yes
checking for vasnprintf... no
checking for wcrtomb... yes
checking for wcwidth... yes
checking for nl_langinfo and CODESET... yes
checking for a traditional french locale... none
checking whether byte ordering is bigendian... (cached) no
checking whether // is distinct from /... unknown, assuming no
checking if environ is properly declared... yes
checking for complete errno.h... yes
checking whether strerror_r is declared... yes
checking for strerror_r... (cached) yes
checking whether strerror_r returns char *... no
checking for sig_atomic_t... (cached) yes
checking for working fcntl.h... cross-compiling
checking for mode_t... yes
checking for mbstate_t... yes
checking whether stdin defaults to large file offsets... yes
checking whether fseeko is declared... yes
checking for fseeko... yes
checking whether stat file-mode macros are broken... no
checking for nlink_t... yes
checking whether fchmodat is declared without a macro... yes
checking whether fstat is declared without a macro... yes
checking whether fstatat is declared without a macro... yes
checking whether futimens is declared without a macro... yes
checking whether lchmod is declared without a macro... yes
checking whether lstat is declared without a macro... yes
checking whether mkdirat is declared without a macro... yes
checking whether mkfifo is declared without a macro... yes
checking whether mkfifoat is declared without a macro... yes
checking whether mknod is declared without a macro... yes
checking whether mknodat is declared without a macro... yes
checking whether stat is declared without a macro... yes
checking whether utimensat is declared without a macro... yes
checking whether ftello is declared... yes
checking for ftello... yes
checking whether ftello works... guessing yes
checking whether getaddrinfo is declared without a macro... yes
checking whether freeaddrinfo is declared without a macro... yes
checking whether gai_strerror is declared without a macro... yes
checking whether getnameinfo is declared without a macro... yes
checking for library containing gethostbyname... none required
checking for gethostbyname... yes
checking for library containing getservbyname... none required
checking for getservbyname... yes
checking for library containing inet_ntop... none required
checking whether inet_ntop is declared... yes
checking for IPv4 sockets... yes
checking for IPv6 sockets... yes
checking whether getdelim is declared... yes
checking whether getdtablesize is declared... yes
checking for uid_t in sys/types.h... yes
checking type of array argument to getgroups... gid_t
checking whether getline is declared... yes
checking for getopt.h... (cached) yes
checking for getopt_long_only... yes
checking whether getopt is POSIX compatible... guessing yes
checking for working GNU getopt function... guessing no
checking whether getpass is declared... yes
checking whether fflush_unlocked is declared... yes
checking whether flockfile is declared... yes
checking whether fputs_unlocked is declared... yes
checking whether funlockfile is declared... yes
checking whether putc_unlocked is declared... yes
checking for struct timeval... yes
checking for wide-enough struct timeval.tv_sec member... yes
checking whether gettimeofday is declared without a macro... yes
checking for iconv... yes
checking for working iconv... guessing yes
checking for iconv declaration... 
         extern size_t iconv (iconv_t cd, char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t *outbytesleft);
checking whether limits.h has ULLONG_WIDTH etc.... no
checking for wint_t... yes
checking whether wint_t is too small... no
checking for unsigned long long int... yes
checking for long long int... yes
checking whether stdint.h conforms to C99... guessing yes
checking whether stdint.h predates C++11... no
checking whether stdint.h has UINTMAX_WIDTH etc.... no
checking whether imaxabs is declared without a macro... yes
checking whether imaxdiv is declared without a macro... yes
checking whether strtoimax is declared without a macro... yes
checking whether strtoumax is declared without a macro... yes
checking for inttypes.h... (cached) yes
checking whether the inttypes.h PRIxNN macros are broken... no
checking whether iswcntrl works... guessing yes
checking for towlower... yes
checking for wctype_t... yes
checking for wctrans_t... yes
checking whether wctype is declared without a macro... yes
checking whether iswctype is declared without a macro... yes
checking whether wctrans is declared without a macro... yes
checking whether towctrans is declared without a macro... yes
checking whether included libunistring is requested... yes
checking whether getc_unlocked is declared... yes
checking whether we are using the GNU C Library >= 2.1 or uClibc... no
checking for wchar_t... yes
checking for max_align_t... yes
checking whether NULL can be used in arbitrary expressions... yes
checking whether imported symbols can be declared weak... guessing yes
checking pthread.h usability... yes
checking pthread.h presence... yes
checking for pthread.h... yes
checking for pthread_kill in -lpthread... yes
checking for multithread API to use... posix
checking whether lstat correctly handles trailing slash... guessing yes
checking for a sed that does not truncate output... /home/test/openwrt/staging_dir/host/bin/sed
checking whether malloc, realloc, calloc are POSIX compliant... yes
checking for stdlib.h... (cached) yes
checking for GNU libc compatible malloc... (cached) yes
checking for a traditional japanese locale... none
checking for a transitional chinese locale... none
checking for a french Unicode locale... none
checking for mmap... yes
checking for MAP_ANONYMOUS... yes
checking whether memchr works... guessing no
checking whether memrchr is declared... yes
checking whether <limits.h> defines MIN and MAX... no
checking whether <sys/param.h> defines MIN and MAX... yes
checking whether time_t is signed... yes
checking whether alarm is declared... yes
checking for working mktime... guessing no
checking whether <sys/select.h> is self-contained... yes
checking whether pselect is declared without a macro... yes
checking whether select is declared without a macro... yes
checking for library containing setsockopt... none needed
checking whether select supports a 0 argument... guessing yes
checking whether select detects invalid fds... guessing yes
checking for O_CLOEXEC... yes
checking for promoted mode_t type... mode_t
checking for library containing posix_spawn... none required
checking for posix_spawn... yes
checking whether posix_spawn works... guessing yes
checking whether posix_spawnattr_setschedpolicy is supported... yes
checking whether posix_spawnattr_setschedparam is supported... yes
checking for sigset_t... yes
checking for SIGPIPE... yes
checking whether C symbols are prefixed with underscore at the linker level... no
checking whether snprintf returns a byte count as in C99... guessing yes
checking whether snprintf is declared... yes
checking whether strdup is declared... yes
checking whether strerror(0) succeeds... guessing yes
checking for strerror_r with POSIX signature... yes
checking whether strerror_r works... guessing yes
checking whether strerror_r is declared... (cached) yes
checking whether ffsl is declared without a macro... yes
checking whether ffsll is declared without a macro... yes
checking whether memmem is declared without a macro... yes
checking whether mempcpy is declared without a macro... yes
checking whether memrchr is declared without a macro... yes
checking whether rawmemchr is declared without a macro... no
checking whether stpcpy is declared without a macro... yes
checking whether stpncpy is declared without a macro... yes
checking whether strchrnul is declared without a macro... yes
checking whether strdup is declared without a macro... yes
checking whether strncat is declared without a macro... yes
checking whether strndup is declared without a macro... yes
checking whether strnlen is declared without a macro... yes
checking whether strpbrk is declared without a macro... yes
checking whether strsep is declared without a macro... yes
checking whether strcasestr is declared without a macro... yes
checking whether strtok_r is declared without a macro... yes
checking whether strerror_r is declared without a macro... yes
checking whether strsignal is declared without a macro... yes
checking whether strverscmp is declared without a macro... yes
checking whether ffs is declared without a macro... yes
checking whether strcasecmp is declared without a macro... yes
checking whether strncasecmp is declared without a macro... yes
checking whether strndup is declared... (cached) yes
checking whether strnlen is declared... (cached) yes
checking for struct tm.tm_gmtoff... yes
checking whether strtok_r is declared... (cached) yes
checking for struct timespec in <time.h>... yes
checking whether clearerr_unlocked is declared... yes
checking whether feof_unlocked is declared... yes
checking whether ferror_unlocked is declared... yes
checking whether fgets_unlocked is declared... yes
checking whether fputc_unlocked is declared... yes
checking whether fread_unlocked is declared... yes
checking whether fwrite_unlocked is declared... yes
checking whether getchar_unlocked is declared... yes
checking whether putchar_unlocked is declared... yes
checking whether the utimes function works... guessing no
checking for inttypes.h... yes
checking for stdint.h... yes
checking for intmax_t... yes
checking where to find the exponent in a 'double'... unknown
checking for snprintf... (cached) yes
checking for strnlen... yes
checking for wcslen... yes
checking for wcsnlen... yes
checking for mbrtowc... (cached) yes
checking for wcrtomb... (cached) yes
checking whether _snprintf is declared... no
checking whether vsnprintf is declared... yes
checking for alloca as a compiler built-in... yes
checking whether inet_ntop is declared without a macro... yes
checking whether inet_pton is declared without a macro... yes
checking whether btowc(0) is correct... guessing yes
checking whether btowc(EOF) is correct... guessing yes
checking for __builtin_expect... yes
checking for library containing clock_gettime... none required
checking for clock_gettime... yes
checking for clock_settime... yes
checking whether // is distinct from /... (cached) unknown, assuming no
checking whether dup2 works... guessing yes
checking for error_at_line... no
checking whether fcntl handles F_DUPFD correctly... guessing yes
checking whether fcntl understands F_DUPFD_CLOEXEC... needs runtime check
checking whether fcntl is declared without a macro... yes
checking whether openat is declared without a macro... yes
checking for flexible array member... yes
checking whether conversion from 'int' to 'long double' works... guessing yes
checking for working POSIX fnmatch... guessing no
checking whether isblank is declared... yes
checking whether fopen recognizes a trailing slash... guessing yes
checking for fseeko... (cached) yes
checking for ftello... (cached) yes
checking whether ftello works... (cached) guessing yes
checking whether futimens works... guessing yes
checking how to do getaddrinfo, freeaddrinfo and getnameinfo... checking for library containing getaddrinfo... none required
checking for getaddrinfo... yes
checking whether gai_strerror is declared... (cached) yes
checking whether gai_strerrorA is declared... no
checking for gai_strerror with POSIX signature... yes
checking for struct sockaddr.sa_len... no
checking whether getaddrinfo is declared... (cached) yes
checking whether freeaddrinfo is declared... (cached) yes
checking whether getnameinfo is declared... (cached) yes
checking for struct addrinfo... yes
checking for working getdelim function... guessing no
checking for flockfile... yes
checking for funlockfile... yes
checking whether getc_unlocked is declared... (cached) yes
checking whether getdtablesize works... guessing yes
checking for getgroups... yes
checking for working getgroups... guessing yes
checking whether getgroups handles negative values... guessing yes
checking for getline... yes
checking for working getline function... guessing no
checking whether __fsetlocking is declared... yes
checking whether program_invocation_name is declared... yes
checking whether program_invocation_short_name is declared... yes
checking whether __argv is declared... no
checking whether gettimeofday clobbers localtime buffer... guessing no
checking for gettimeofday with POSIX signature... yes
checking for group_member... no
checking for library containing gethostbyname... (cached) none required
checking for gethostbyname... (cached) yes
checking for library containing inet_ntop... (cached) none required
checking whether inet_ntop is declared... (cached) yes
checking whether the compiler generally respects inline... yes
checking whether INT32_MAX < INTMAX_MAX... yes
checking whether INT64_MAX == LONG_MAX... no
checking whether UINT32_MAX < UINTMAX_MAX... yes
checking whether UINT64_MAX == ULONG_MAX... no
checking for ioctl... yes
checking for ioctl with POSIX signature... yes
checking whether iswblank is declared... yes
checking whether langinfo.h defines CODESET... yes
checking whether langinfo.h defines T_FMT_AMPM... yes
checking whether langinfo.h defines ERA... yes
checking whether langinfo.h defines YESEXPR... yes
checking whether nl_langinfo is declared without a macro... yes
checking whether to use the included libunistring... yes
checking whether link obeys POSIX... guessing yes
checking whether locale.h conforms to POSIX:2001... yes
checking whether struct lconv is properly defined... yes
checking whether setlocale is declared without a macro... yes
checking whether duplocale is declared without a macro... yes
checking for pthread_rwlock_t... yes
checking whether pthread_rwlock_rdlock prefers a writer to a reader... guessing yes
checking whether lseek detects pipes... yes
checking whether mbrtowc handles incomplete characters... guessing yes
checking whether mbrtowc works as well as mbtowc... guessing yes
checking whether mbrtowc handles a NULL pwc argument... guessing yes
checking whether mbrtowc handles a NULL string argument... guessing yes
checking whether mbrtowc has a correct return value... guessing yes
checking whether mbrtowc returns 0 when parsing a NUL character... guessing yes
checking whether mbrtowc works on empty input... guessing no
checking whether the C locale is free of encoding errors... guessing no
checking whether mbrtowc handles incomplete characters... (cached) guessing yes
checking whether mbrtowc works as well as mbtowc... (cached) guessing yes
checking whether mbrtowc handles incomplete characters... (cached) guessing yes
checking whether mbrtowc works as well as mbtowc... (cached) guessing yes
checking whether mbsrtowcs works... guessing yes
checking bp-sym.h usability... no
checking bp-sym.h presence... no
checking for bp-sym.h... no
checking for memrchr... yes
checking whether mkdir handles trailing slash... guessing yes
checking whether mkdir handles trailing dot... guessing yes
checking for working mkstemp... guessing yes
checking for __mktime_internal... no
checking for library containing nanosleep... none required
checking for working nanosleep... guessing no (mishandles large arguments)
checking whether getaddrinfo is declared without a macro... (cached) yes
checking whether freeaddrinfo is declared without a macro... (cached) yes
checking whether gai_strerror is declared without a macro... (cached) yes
checking whether getnameinfo is declared without a macro... (cached) yes
checking whether <netinet/in.h> is self-contained... yes
checking whether YESEXPR works... guessing yes
checking whether open recognizes a trailing slash... guessing yes
checking whether posix_spawn_file_actions_addclose works... guessing yes
checking whether posix_spawn_file_actions_adddup2 works... guessing yes
checking whether posix_spawn_file_actions_addopen works... guessing yes
checking for raise... yes
checking for sigprocmask... yes
checking for rawmemchr... no
checking for working re_compile_pattern... guessing no
checking libintl.h usability... yes
checking libintl.h presence... yes
checking for libintl.h... yes
checking whether isblank is declared... (cached) yes
checking for __secure_getenv... no
checking for issetugid... yes
checking whether select supports a 0 argument... (cached) guessing yes
checking whether select detects invalid fds... (cached) guessing yes
checking for library containing getservbyname... (cached) none required
checking for getservbyname... (cached) yes
checking for struct sigaction.sa_sigaction... yes
checking for volatile sig_atomic_t... yes
checking for sighandler_t... yes
checking whether pthread_sigmask is declared without a macro... yes
checking whether sigaction is declared without a macro... yes
checking whether sigaddset is declared without a macro... yes
checking whether sigdelset is declared without a macro... yes
checking whether sigemptyset is declared without a macro... yes
checking whether sigfillset is declared without a macro... yes
checking whether sigismember is declared without a macro... yes
checking whether sigpending is declared without a macro... yes
checking whether sigprocmask is declared without a macro... yes
checking for sigprocmask... (cached) yes
checking for stdint.h... (cached) yes
checking for SIZE_MAX... yes
checking for snprintf... (cached) yes
checking whether snprintf respects a size of 1... guessing yes
checking whether printf supports POSIX/XSI format strings with positions... guessing yes
checking for socklen_t... yes
checking for posix_spawnattr_t... yes
checking for posix_spawn_file_actions_t... yes
checking whether posix_spawn is declared without a macro... yes
checking whether posix_spawnp is declared without a macro... yes
checking whether posix_spawnattr_init is declared without a macro... yes
checking whether posix_spawnattr_destroy is declared without a macro... yes
checking whether posix_spawnattr_getsigdefault is declared without a macro... yes
checking whether posix_spawnattr_setsigdefault is declared without a macro... yes
checking whether posix_spawnattr_getsigmask is declared without a macro... yes
checking whether posix_spawnattr_setsigmask is declared without a macro... yes
checking whether posix_spawnattr_getflags is declared without a macro... yes
checking whether posix_spawnattr_setflags is declared without a macro... yes
checking whether posix_spawnattr_getpgroup is declared without a macro... yes
checking whether posix_spawnattr_setpgroup is declared without a macro... yes
checking whether posix_spawnattr_getschedpolicy is declared without a macro... yes
checking whether posix_spawnattr_setschedpolicy is declared without a macro... yes
checking whether posix_spawnattr_getschedparam is declared without a macro... yes
checking whether posix_spawnattr_setschedparam is declared without a macro... yes
checking whether posix_spawn_file_actions_init is declared without a macro... yes
checking whether posix_spawn_file_actions_destroy is declared without a macro... yes
checking whether posix_spawn_file_actions_addopen is declared without a macro... yes
checking whether posix_spawn_file_actions_addclose is declared without a macro... yes
checking whether posix_spawn_file_actions_adddup2 is declared without a macro... yes
checking for ssize_t... yes
checking whether stat handles trailing slashes on files... guessing yes
checking for struct stat.st_atim.tv_nsec... yes
checking whether struct stat.st_atim is of type struct timespec... yes
checking for struct stat.st_birthtimespec.tv_nsec... no
checking for struct stat.st_birthtimensec... no
checking for struct stat.st_birthtim.tv_nsec... no
checking for working stdalign.h... yes
checking for max_align_t... (cached) yes
checking whether NULL can be used in arbitrary expressions... (cached) yes
checking which flavor of printf attribute matches inttypes macros... system
checking whether dprintf is declared without a macro... yes
checking whether fpurge is declared without a macro... no
checking whether fseeko is declared without a macro... yes
checking whether ftello is declared without a macro... yes
checking whether getdelim is declared without a macro... yes
checking whether getline is declared without a macro... yes
checking whether gets is declared without a macro... no
checking whether pclose is declared without a macro... yes
checking whether popen is declared without a macro... yes
checking whether renameat is declared without a macro... yes
checking whether snprintf is declared without a macro... yes
checking whether tmpfile is declared without a macro... yes
checking whether vdprintf is declared without a macro... yes
checking whether vsnprintf is declared without a macro... yes
checking whether _Exit is declared without a macro... yes
checking whether atoll is declared without a macro... yes
checking whether canonicalize_file_name is declared without a macro... no
checking whether getloadavg is declared without a macro... yes
checking whether getsubopt is declared without a macro... yes
checking whether grantpt is declared without a macro... yes
checking whether initstate is declared without a macro... yes
checking whether initstate_r is declared without a macro... no
checking whether mkdtemp is declared without a macro... yes
checking whether mkostemp is declared without a macro... yes
checking whether mkostemps is declared without a macro... yes
checking whether mkstemp is declared without a macro... yes
checking whether mkstemps is declared without a macro... yes
checking whether posix_openpt is declared without a macro... yes
checking whether ptsname is declared without a macro... yes
checking whether ptsname_r is declared without a macro... yes
checking whether qsort_r is declared without a macro... no
checking whether random is declared without a macro... yes
checking whether random_r is declared without a macro... no
checking whether reallocarray is declared without a macro... no
checking whether realpath is declared without a macro... yes
checking whether rpmatch is declared without a macro... no
checking whether secure_getenv is declared without a macro... no
checking whether setenv is declared without a macro... yes
checking whether setstate is declared without a macro... yes
checking whether setstate_r is declared without a macro... no
checking whether srandom is declared without a macro... yes
checking whether srandom_r is declared without a macro... no
checking whether strtod is declared without a macro... yes
checking whether strtoll is declared without a macro... yes
checking whether strtoull is declared without a macro... yes
checking whether unlockpt is declared without a macro... yes
checking whether unsetenv is declared without a macro... yes
checking for strcasecmp... yes
checking for strncasecmp... yes
checking whether strncasecmp is declared... (cached) yes
checking for strchrnul... yes
checking whether strchrnul works... guessing yes
checking for working strerror function... guessing yes
checking for working strndup... guessing yes
checking for working strnlen... yes
checking for strpbrk... yes
checking for strtok_r... yes
checking whether strtok_r works... guessing no
checking for strtol... yes
checking for strtoll... yes
checking whether symlink handles trailing slash correctly... guessing yes
checking whether flock is declared without a macro... yes
checking whether <sys/ioctl.h> declares ioctl... yes
checking whether ioctl is declared without a macro... yes
checking for nlink_t... (cached) yes
checking whether fchmodat is declared without a macro... (cached) yes
checking whether fstat is declared without a macro... (cached) yes
checking whether fstatat is declared without a macro... (cached) yes
checking whether futimens is declared without a macro... (cached) yes
checking whether lchmod is declared without a macro... (cached) yes
checking whether lstat is declared without a macro... (cached) yes
checking whether mkdirat is declared without a macro... (cached) yes
checking whether mkfifo is declared without a macro... (cached) yes
checking whether mkfifoat is declared without a macro... (cached) yes
checking whether mknod is declared without a macro... (cached) yes
checking whether mknodat is declared without a macro... (cached) yes
checking whether stat is declared without a macro... (cached) yes
checking whether utimensat is declared without a macro... (cached) yes
checking whether waitpid is declared without a macro... yes
checking whether localtime_r is declared... yes
checking whether localtime_r is compatible with its POSIX signature... yes
checking whether chdir is declared without a macro... yes
checking whether chown is declared without a macro... yes
checking whether dup is declared without a macro... yes
checking whether dup2 is declared without a macro... yes
checking whether dup3 is declared without a macro... yes
checking whether environ is declared without a macro... yes
checking whether euidaccess is declared without a macro... yes
checking whether faccessat is declared without a macro... yes
checking whether fchdir is declared without a macro... yes
checking whether fchownat is declared without a macro... yes
checking whether fdatasync is declared without a macro... yes
checking whether fsync is declared without a macro... yes
checking whether ftruncate is declared without a macro... yes
checking whether getcwd is declared without a macro... yes
checking whether getdomainname is declared without a macro... yes
checking whether getdtablesize is declared without a macro... yes
checking whether getgroups is declared without a macro... yes
checking whether gethostname is declared without a macro... yes
checking whether getlogin is declared without a macro... yes
checking whether getlogin_r is declared without a macro... yes
checking whether getpagesize is declared without a macro... yes
checking whether getusershell is declared without a macro... yes
checking whether setusershell is declared without a macro... yes
checking whether endusershell is declared without a macro... yes
checking whether group_member is declared without a macro... no
checking whether isatty is declared without a macro... yes
checking whether lchown is declared without a macro... yes
checking whether link is declared without a macro... yes
checking whether linkat is declared without a macro... yes
checking whether lseek is declared without a macro... yes
checking whether pipe is declared without a macro... yes
checking whether pipe2 is declared without a macro... yes
checking whether pread is declared without a macro... yes
checking whether pwrite is declared without a macro... yes
checking whether readlink is declared without a macro... yes
checking whether readlinkat is declared without a macro... yes
checking whether rmdir is declared without a macro... yes
checking whether sethostname is declared without a macro... yes
checking whether sleep is declared without a macro... yes
checking whether symlink is declared without a macro... yes
checking whether symlinkat is declared without a macro... yes
checking whether truncate is declared without a macro... yes
checking whether ttyname_r is declared without a macro... yes
checking whether unlink is declared without a macro... yes
checking whether unlinkat is declared without a macro... yes
checking whether usleep is declared without a macro... yes
checking whether unlink honors trailing slashes... guessing yes
checking whether unlink of a parent directory fails as it should... guessing yes
checking whether utime is declared without a macro... yes
checking for ptrdiff_t... yes
checking for vasprintf... yes
checking for vsnprintf... (cached) yes
checking whether snprintf respects a size of 1... (cached) guessing yes
checking whether printf supports POSIX/XSI format strings with positions... (cached) guessing yes
checking for waitid... yes
checking whether btowc is declared without a macro... yes
checking whether wctob is declared without a macro... yes
checking whether mbsinit is declared without a macro... yes
checking whether mbrtowc is declared without a macro... yes
checking whether mbrlen is declared without a macro... yes
checking whether mbsrtowcs is declared without a macro... yes
checking whether mbsnrtowcs is declared without a macro... yes
checking whether wcrtomb is declared without a macro... yes
checking whether wcsrtombs is declared without a macro... yes
checking whether wcsnrtombs is declared without a macro... yes
checking whether wcwidth is declared without a macro... yes
checking whether wmemchr is declared without a macro... yes
checking whether wmemcmp is declared without a macro... yes
checking whether wmemcpy is declared without a macro... yes
checking whether wmemmove is declared without a macro... yes
checking whether wmemset is declared without a macro... yes
checking whether wcslen is declared without a macro... yes
checking whether wcsnlen is declared without a macro... yes
checking whether wcscpy is declared without a macro... yes
checking whether wcpcpy is declared without a macro... yes
checking whether wcsncpy is declared without a macro... yes
checking whether wcpncpy is declared without a macro... yes
checking whether wcscat is declared without a macro... yes
checking whether wcsncat is declared without a macro... yes
checking whether wcscmp is declared without a macro... yes
checking whether wcsncmp is declared without a macro... yes
checking whether wcscasecmp is declared without a macro... yes
checking whether wcsncasecmp is declared without a macro... yes
checking whether wcscoll is declared without a macro... yes
checking whether wcsxfrm is declared without a macro... yes
checking whether wcsdup is declared without a macro... yes
checking whether wcschr is declared without a macro... yes
checking whether wcsrchr is declared without a macro... yes
checking whether wcscspn is declared without a macro... yes
checking whether wcsspn is declared without a macro... yes
checking whether wcspbrk is declared without a macro... yes
checking whether wcsstr is declared without a macro... yes
checking whether wcstok is declared without a macro... yes
checking whether wcswidth is declared without a macro... yes
checking whether wcsftime is declared without a macro... yes
checking whether mbrtowc handles incomplete characters... (cached) guessing yes
checking whether mbrtowc works as well as mbtowc... (cached) guessing yes
checking whether wcrtomb return value is correct... guessing yes
checking whether iswcntrl works... (cached) guessing yes
checking for towlower... (cached) yes
checking for wctype_t... (cached) yes
checking for wctrans_t... (cached) yes
checking whether wctype is declared without a macro... (cached) yes
checking whether iswctype is declared without a macro... (cached) yes
checking whether wctrans is declared without a macro... (cached) yes
checking whether towctrans is declared without a macro... (cached) yes
checking whether wcwidth is declared... (cached) yes
checking whether wcwidth works reasonably in UTF-8 locales... guessing yes
checking for stdint.h... (cached) yes
checking for getpagesize... yes
checking for working mmap... no
checking for _LARGEFILE_SOURCE value needed for large files... (cached) no
checking for strptime... (cached) yes
checking for timegm... (cached) yes
checking for vsnprintf... (cached) yes
checking for vasprintf... (cached) yes
checking for drand48... yes
checking for pathconf... yes
checking for strtoll... (cached) yes
checking for usleep... yes
checking for ftello... yes
checking for sigblock... no
checking for sigsetjmp... yes
checking for memrchr... (cached) yes
checking for wcwidth... (cached) yes
checking for mbtowc... yes
checking for sleep... yes
checking for symlink... (cached) yes
checking for utime... (cached) yes
checking for strlcpy... yes
checking for random... yes
checking for fmemopen... yes
checking pkg-config is at least version 0.9.0... yes
checking for ZLIB... yes
checking for dlopen in -ldl... yes
checking for libssl... no
configure: error: --with-ssl=openssl was given, but SSL is not available.
make[3]: *** [Makefile:104: /home/test/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/wget-ssl/wget-1.19.5/.configured_68b329da9893e34099c7d8ad5cb9c940] Error 1
make[3]: Leaving directory '/home/test/openwrt/feeds/packages/net/wget'
time: package/feeds/packages/wget/ssl/compile#16.63#8.90#21.77
make[2]: *** [package/Makefile:109: package/feeds/packages/wget/compile] Error 2
make[2]: Leaving directory '/home/test/openwrt'
make[1]: *** [package/Makefile:103: /home/test/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/test/openwrt'
make: *** [/home/test/openwrt/include/toplevel.mk:218: world] Error 2
test@test-virtual-machine:~/openwrt$ nano target/
Config.in     imagebuilder/ linux/        Makefile      sdk/          toolchain/    
test@test-virtual-machine:~/openwrt$ nano feeds/packages
packages/             packages.index        packages.targetindex  packages.tmp/         
test@test-virtual-machine:~/openwrt$ nano feeds/packages/
admin/           devel/           .github/         .keys/           LICENSE          net/             .travis_do.sh    
.circleci/       fonts/           ipv6/            lang/            mail/            README.md        .travis.yml      
CONTRIBUTING.md  .git/            kernel/          libs/            multimedia/      sound/           utils/           
test@test-virtual-machine:~/openwrt$ nano feeds/packages/net/wget/patches/010-openssl-Do-not-use-engines-when-OpenSSL-does-not-sup.patch 
```

**Comment by yousong** [comment]
@frymezim , you will need to find out why wget configure cannot find `libssl`, probably by checking config.log in the build dir.

**Comment by yousong** [comment]
I'd like to also note that the 3 packages in the original report now has support for building without openssl deprecated apis.  I think @neheb does most of the work here, thank you.
